robottelo/config/facade.py

```diff
@@ -0,0 +1,311 @@
+import logging
+import os
+from functools import reduce
+from urllib.parse import urljoin
+from urllib.parse import urlunsplit
+
+from wrapt import CallableObjectProxy
+
+from robottelo.config.base import get_project_root
+from robottelo.config.base import INIReader
+from robottelo.config.base import SETTINGS_FILE_NAME
+
+logger = logging.getLogger('robottelo.settings.proxy')
+
+WRAPPER_EXCEPTIONS = (
+    'server.hostname',
+    'server.ssh_key',
+    'server.ssh_password',
+    'server.port',
+    'server.scheme',
+    'azurerm.azure_region',
+    'azurerm.client_id',
+    'azurerm.client_secret',
+    'azurerm.password',
+    'azurerm.ssh_pub_key',
+    'azurerm.subscription_id',
+    'azurerm.tenant_id',
+    'azurerm.username',
+    'compute_resources.libvirt_hostname',
```

Comment:
with the addition of a new broker setting, you'll like need to add that in here.

Reply:
From what I see, `broker_directory` is read from settings and immediately put in environmental variable. Nothing in robottelo tries to read anything from `settings.broker` object - I assume broker uses this environmental variable.

This reading and variable setting is done inside legacy Settings class, before facade is constructed. Since nothing tries to fetch this value through facade, there is no need for facade to treat this setting as exception.

And even then - facade needs exceptions only for settings that are serialized to JSON (as Python's `json` module can serialize only basic data structures and does not allow objects to define how they should be serialized), or are passed to some picky modules that look into actual variable type. No need to put every single setting into exceptions list.